### DISCUSSION TITLE
[Feature] Make an album available offline

### DISCUSSION THREAD

**Discussion by t18n** [immich-app/immich#7627] [Feature Request]
### The feature

Sometimes, I have a new album that I want to show to another person, but if a server is down or on a slow network request, it can ruin the experience. Would be awesome if we can download an album in advance. Alternatively, you can add an option to "save and persist" an album to the native photo library (on iOS Photos app) so it is available there.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by meldalinn** [comment]
Would also like to see this feature

**Comment by NicholasFlamy** [comment]
I feel like this would be part of a major overhaul of the backup system and would especially fit well with the eventual planned feature for automaticly removing backed-up photos from the device to free up space.

**Reply by ChiNoel-osu** [reply]
Indeed, freeing up space on mobile is huge. Sort of like how iCloud works where it stores previews on device and the full size on cloud.

**Comment by jantari** [comment]
Yes I totally need this for shared albums. My own pictures I already have available offline, because I took them, but the pictures other users added to their or my shared album aren´t available to me when I'm not at home in my WiFi where I can reach my immich instance.

It would be great to be able to cache / download a whole shared album offline with a few clicks.

**Comment by aero31aero** [comment]
This could be entirely separate from the existing backup pipeline. Just do what Spotify and Youtube Music etc do for offline playlists, and download a slightly lower quality variant of those photos onto the phone storage.

Those won't be visible in other gallery apps; only in Immich's "Offline Albums" section. This would eliminate some kind of backup looks where you could accidentally set an existing album to offline, and then immich starts backing it up again.

**Comment by SHU-red** [comment]
I had an idea for a cool feature request which would be very handy for me.

As a well-behaving immich & github user i found this discussion and wanted to add my idea to this discussion:

What i like to do (and whats especially much more safe on a self-hosted solution like immich) is:
- I photograph/scan important documents of mine
- These are e.g. my ID, my international ID and even my vehicle registration documents
- I pack these photos into an album or tag them to quickly find them as needed (e.g. for ordering parts for my (or my families) cars)
- I need these documents always available &rarr; also offline

Now with the absolutely awesome Tags-feature of immich im absolutely pumped in creating my own private "document-structure" by being able to hierarchic sort them, search them and carry them with me

So i would like to extend the "offline-sync" feature to be able to e.g. mark
- albums
- persons
- tags
- ...

to be synced offline to my mobile devices

During traveling, being offline or in remote countries this would be a nice feature for me

**Reply by aero31aero** [reply]
You need paperless-ng or bitwarden, not Immich. Bitwarden for identity docs and paperless for receipts and such.

**Reply by SHU-red** [reply]
Hi @aero31aero,
thank you taking the time

I already host
- paperless ngx: Is it possible to sync certain documents offline? I dont think so ...
- bitwarden: Nice idea. Is it possible to keep/upload files/images/pdfs?
- nextcloud: This would be the right place i think but the UI for immich is much more appealing to me and the tagging function would go much nicer than files in folders

I appreciate your suggestions

And i also admit, that i am somehow biased to use my "image-app" coming from Google Photos and using GPhotos "details" function to add searchable words to images

I will think about it, but anyways it would be a cool and powerful feature for immich.

Just in case: maybe i did not clarify that i want to have all of the above mentioned documents as images (not pdfs or something) to be able to always show my documents quick and easy

**Reply by aero31aero** [reply]
Maybe I'm in a different universe but mixing even screenshots with "photos" feels weird. Yes, Bitwarden has 1GB of storage for documents/files. It takes images too. 

Also, you can also self-host vaultwarden if you wanna have unlimited Bitwarden storage.

And, IIRC the paperless app on FDroid lets you download docs offline, but I'm not sure. You'd have to check that one.

**Reply by NicholasFlamy** [reply]
> paperless app on FDroid lets you download docs offline

So you can download the stuff but it'll just be a file on your phone, offline management and viewing from the paperless app is a feature request: https://github.com/astubenbord/paperless-mobile/discussions/102

**Comment by Durnalicious** [comment]
Hi all,

I wanted to add my support on this feature request as well, but wanted to be a bit more specific about how I believe the implementation for this feature request should look.

Specifically, I believe the option to download or make certain photos or albums "available offline" should remain within the smartphone application.

The use case that I have for this, is being able to keep certain albums available locally on my device while traveling or otherwise not having network connectivity, but still being able to show someone an album (e.g. Talking to someone on a plane, we discuss our pets, and I'd love to show them some photos of my dog. Hard to do at 35,000 feet!)

I think for implementation, realistically the way that Nextcloud does this (and OneDrive as well, really) is a pretty good implementation of this. You can select individual files or folders (and in the case of Immich, it would be albums) and select "Make Always Available Locally" which stores them locally on the device. The files are still shown within the Nextcloud app, which helps keep it within "the workflow" of the software.

In the future if you no longer need them locally on your device, you can select "Free up local space" and it deletes the local copy on the device. The original remains on the Immich server.

I think an implementation similar to this would be fantastic. Mark an album as "available locally", the album is viewable within the Immich smart app so it stays within the established workflow (and doesn't clutter other system gallery/photo apps!), and when no longer needed the space can be freed up.

I've attached a photo of what this looks like within the Nextcloud client on Windows:

![Nextcloud Make Available Locally](https://github.com/user-attachments/assets/45132a9d-8dc4-4084-9ab1-40f2ed94cc52)

as well as on Android (option is named differently on Android as "Sync"):

![Nextcloud Android App Sync](https://github.com/user-attachments/assets/77be4cd4-bcba-49f5-af04-842bd8c7886c)

After syncing, you can see what is locally available from the hamburger menu, and going to "On device":

![Nextcloud Android App On Device](https://github.com/user-attachments/assets/7a0f8ff1-94a1-42e6-a168-affb177aea83)

All in all, I think an implementation that behaves similar to Nextcloud's solution would be more than sufficient for 99% of users (myself included). I greatly appreciate the work you (the Immich Dev team!) do, and am pleasantly excited every time there is a new change-log entry and update to see what's been added.

Thank you for the consideration on this request, and for all of your continued hard work!

-Durnalicious

**Comment by crashmit** [comment]
I would like to add the possibility to have my whole library offline - or just albums depending on my requirements.

**Reply by anaxmedia** [reply]
This would fit my use case as well. I'd like for Immich to replace my photos app. I keep everything sync'd to an external library and would love to have access to everything locally (cached previews) on my phone.

**Comment by jomir1000** [comment]
offline albums would be awesome. since some of the pictures in (shared) albums are already cached (i guess they are getting cached when you are online and watch them), maybe an option like "cache whole album" would suffice.

**Comment by TiagoDM** [comment]
I think it would be a must to be able to download whole albums to the device, since i think it's not possible to do that at this point. From the web interface it is possible to download albums, but not in android app. Please do add that feature. Cheers.

**Reply by NicholasFlamy** [reply]
Make sure to hit the upvote button. 
![image](https://github.com/user-attachments/assets/ce652cec-0baa-48ea-9e38-5fb47f57fbcf)

**Reply by TiagoDM** [reply]
Already did. thanks.

**Comment by joelhinch** [comment]
This is a dealbreaker for me. Need this.

**Comment by CodingGhost** [comment]
couldnt there be a really easy solution? you can already do bulk downloads, the only issue is that its not possible to select a full album, you have to select each day of photos separately. With the little change of a select all button, the issue would be mostly solved I think?

**Reply by jantari** [reply]
The problem with that is that you would have to come in every day and re-download-all to have new pictures available, that'd be very tedious. Doesn't really solve the problem imo

**Reply by Durnalicious** [reply]
The Immich mobile app already has functionality to determine if an asset is local only and has not been backed up, local and has been backed up, or cloud only built in. This is clear by the [icons in the mobile application](https://immich.app/docs/FAQ/#what-is-the-difference-between-the-cloud-icons-on-the-mobile-app)

I have a sneaking suspicion that this functionality could be utilized in order to implement [my suggestion above](https://github.com/immich-app/immich/discussions/7627#discussioncomment-11462767).

Especially when you consider that the "Library" tab in the mobile application has a default folder comprised of assets "On this device".

That's my hope, at least! In the meantime, I'm enjoying the rest of the improvements the Immich team have been pushing.

**Reply by CodingGhost** [reply]
but I still think a select all button would help. If you just add photos from your phone, downloading them all once will be enough as every new photo is on your phone anyways. Also there are some albums I just want on my phone locally, that dont receive any new pictures. I just think a select all option should be so easy to implement that its not that important if it doesnt solve all usecases.

**Reply by Durnalicious** [reply]
I believe you and I are talking about the same concept.

> Specifically, I believe the option to download or make certain photos or albums "available offline" should remain within the smartphone application.

So we're in agreement - I think the feature would be beneficial.

**Comment by boronov** [comment]
Hi! Any updates?

**Reply by NicholasFlamy** [reply]
Do you see any updates? If not, the answer is no. Please don't post these kinds of comments.

**Comment by reChords25** [comment]
Also, as t18n described, the Immich app sometimes is very slow when the server is not reachable. One example to get a workaround for this is letting the user manually set how often the app tries to connect to the server and how long the timeout is. In my case, it happens very often that the server being unreachable is expected. When I want to show something to someone in such a situation, it often takes very long for the app to open.

**Comment by JosFas** [comment]
I would love to have "offline albums" by selection. Would be such a nice and fast Feature to have some albums offline.

**Comment by sibaar** [comment]
Second this! Been asking for offline images for a while.

**Comment by jore731** [comment]
I have some heavy albums that I want to keep in High Resolution on immich but would love to have an offline-available copy on my phone. It would be great to be able to make an album available offline with the previews, so it can be visualized and shared in low/medium quality but also downloaded the full HD image when required for printing, posting...
It would be enough to just do it in a one-time effort or with some chron mechanism.
Open to contribute if seems like a good idea!

**Reply by gubbel-alt** [reply]
That's exactly what I want, similar to Apple Photos with the "Optimize iPhone Storage" function.

**Comment by rcarpa** [comment]
Hello everybody. I've spent a bit of time making a fork of the iOS client for my own use. For now I have no intention of seriously maintaining a public version of it. Nevertheless, for those who want to try it, the TestFlight link is in my fork's readme: https://github.com/rcarpa/immich

My honest hope is that a working demo restarts the discussion and the work on this upstream. Nobody enjoys maintaining a fork just for themselves.

This probably interests people following #4466 as well, but I'd rather not cross-post into every thread about these features.
